network/connection.go

```diff
@@ -352,6 +366,13 @@ func NewConnectionFromFirstPacket(pkt packet.Packet) *Connection {
 		Started:                time.Now().Unix(),
 		ProfileRevisionCounter: proc.Profile().RevisionCnt(),
 	}
+
+	// Inherit internal status of profile.
+	if localProfile := proc.Profile().LocalProfile(); localProfile != nil {
+		newConn.Internal = localProfile.Internal
+	}
+
```

Comment:
Now that I see it the third time I start thinking if we should create a dedicated factory function/helper for those inheritance parts. Something like `newConn.inherit(profile)`. Or maybe a helper `NewInhertitedConnection(profile Profile) *Connection`